synapse/storage/databases/main/devices.py

```diff
@@ -342,7 +342,11 @@ def _get_device_updates_by_remote_txn(
             limit: Maximum number of device updates to return
 
         Returns:
-            List: List of device updates
+            List: List of device update tuples:
+                - user_id
+                - device_id
+                - stream_id
+                - opentracing_context
```

Comment:
Could be an attrs/typing.NamedTuple/dataclass here in the future?

Reply:
It appears to be a JSON encoded thing after looking at some other code, but I feel more comfortable leaving it as an opaque clob for the time being